### PR TITLE
[cff] add cff file

### PR DESCRIPTION
**Pull request by dglaeser**
@timokoch, I am setting up some release automation. As a first step this adds a .cff file. Would you mind checking if you are ok with this?

### CODE REVIEW

**Comment by timokoch** [comment]
Should we wait for JOSS and include that as reference publication? Already have a DOI there?

**Comment by timokoch** [comment]
You can also move me to "contributors" instead of "author"

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    family-names: Koch
+    orcid: 'https://orcid.org/0000-0003-4776-5222'
+    affiliation: University of Oslo
+    email: timokoch@math.uio.no
+repository-code: 'https://github.com/dglaeser/gridformat'
+url: 'https://dglaeser.github.io/gridformat/'
+abstract: Header-only C++-Library for grid file I/O
+keywords:
+  - numerical simulation
+  - grid file I/O
```
Comment (timokoch) [inline CITATION.cff:29]:
```suggestion
  - grid file I/O
  - mesh file I/O
```
may be easier to find

**Comment by dglaeser** [comment]
> Should we wait for JOSS and include that as reference publication? Already have a DOI there?

I wanted to set up the automatic release process with the zenodo integration. The github-zenodo integration takes metadata from the .cff file, so I wanted to set that up now s.t. we can generate the zenodo publication required for JOSS. We can add the JOSS paper later on?

**Comment by dglaeser** [comment]
@timokoch, I'll merge this now, you can still comment the .cff file in #225

**Pull request merged**